Bottom: a130d0899b3d1e671d173ee19a640f368c018b4a
Top:    64f0aba74c84b5586ef7fa0e7ce167273aeec466
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-25 13:47:45 +0000

find.c: Remove file-global "IsBad", use PlowsPolygons() return value.

This variable was used to communicate state from the drc_callback(),
however PlowsPolygons() returns the sum of the values returned by its
callback function. As our callback returns 0 on success, 1 on DRC
failure, we know any non-zero return from PlowsPolygons() implies a
DRC problem.


---

diff --git a/src/find.c b/src/find.c
index c2219ee..268b54e 100644
--- a/src/find.c
+++ b/src/find.c
@@ -273,7 +273,6 @@ static void *thing_ptr1, *thing_ptr2, *thing_ptr3;
 static int thing_type;
 static bool User = false;    /* user action causing this */
 static bool drc = false;     /* whether to stop if finding something not found */
-static bool IsBad = false;
 static Cardinal drcerr_count;   /* count of drc errors */
 static Cardinal TotalP, TotalV, NumberOfPads[2];
 static ListType LineList[MAX_LAYER],    /* list of objects to */
@@ -3540,11 +3539,10 @@ doIsBad:
   pcb_drc_violation_free (violation);
   free (object_id_list);
   free (object_type_list);
+
   if (!throw_drc_dialog())
-    {
-      IsBad = true;
-      return 1;
-    }
+    return 1;
+
   IncrementUndoSerialNumber ();
   Undo (true);
   return 0;
@@ -3564,6 +3562,7 @@ DRCAll (void)
   DrcViolationType *violation;
   int tmpcnt;
   int nopastecnt = 0;
+  bool IsBad;
 
   reset_drc_dialog_message();
 
@@ -3638,9 +3637,11 @@ DRCAll (void)
       COPPERLINE_LOOP (PCB->Data);
       {
         /* check line clearances in polygons */
-        PlowsPolygon (PCB->Data, LINE_TYPE, layer, line, drc_callback, NULL);
-        if (IsBad)
-          break;
+        if (PlowsPolygon (PCB->Data, LINE_TYPE, layer, line, drc_callback, NULL))
+          {
+            IsBad = true;
+            break;
+          }
         if (line->Thickness < PCB->minWid)
           {
             AddObjectToFlagUndoList (LINE_TYPE, layer, line, line);
@@ -3680,9 +3681,11 @@ DRCAll (void)
     {
       COPPERARC_LOOP (PCB->Data);
       {
-        PlowsPolygon (PCB->Data, ARC_TYPE, layer, arc, drc_callback, NULL);
-        if (IsBad)
-          break;
+        if (PlowsPolygon (PCB->Data, ARC_TYPE, layer, arc, drc_callback, NULL))
+          {
+            IsBad = true;
+            break;
+          }
         if (arc->Thickness < PCB->minWid)
           {
             AddObjectToFlagUndoList (ARC_TYPE, layer, arc, arc);
@@ -3722,9 +3725,11 @@ DRCAll (void)
     {
       ALLPIN_LOOP (PCB->Data);
       {
-        PlowsPolygon (PCB->Data, PIN_TYPE, element, pin, drc_callback, NULL);
-        if (IsBad)
-          break;
+        if (PlowsPolygon (PCB->Data, PIN_TYPE, element, pin, drc_callback, NULL))
+          {
+            IsBad = true;
+            break;
+          }
         if (!TEST_FLAG (HOLEFLAG, pin) &&
             pin->Thickness - pin->DrillingHole < 2 * PCB->minRing)
           {
@@ -3796,9 +3801,11 @@ DRCAll (void)
     {
       ALLPAD_LOOP (PCB->Data);
       {
-        PlowsPolygon (PCB->Data, PAD_TYPE, element, pad, drc_callback, NULL);
-        if (IsBad)
-          break;
+        if (PlowsPolygon (PCB->Data, PAD_TYPE, element, pad, drc_callback, NULL))
+          {
+            IsBad = true;
+            break;
+          }
         if (pad->Thickness < PCB->minWid)
           {
             AddObjectToFlagUndoList (PAD_TYPE, element, pad, pad);
@@ -3838,9 +3845,11 @@ DRCAll (void)
     {
       VIA_LOOP (PCB->Data);
       {
-        PlowsPolygon (PCB->Data, VIA_TYPE, via, via, drc_callback, NULL);
-        if (IsBad)
-          break;
+        if (PlowsPolygon (PCB->Data, VIA_TYPE, via, via, drc_callback, NULL))
+          {
+            IsBad = true;
+            break;
+          }
         if (!TEST_FLAG (HOLEFLAG, via) &&
             via->Thickness - via->DrillingHole < 2 * PCB->minRing)
           {
